static/js/redirection.js

```diff
@@ -17,6 +17,15 @@ if (path) {
   window.location.href = `${param("path").replace(".md", "")}#${param("q")}`;
 }
 
+const availableIds = [...document.querySelectorAll("[id]")].map(e => e.id)
+const currentHash = window.location.hash.length > 0 ? window.location.hash.substring(1) : '';
+const currentHashIsFound = availableIds.includes(currentHash)
+if (currentHash && !currentHashIsFound) {
+  const alternativeHash = availableIds.find(id => id.replace(/-/g, '') === currentHash);
+  if (alternativeHash)
+    window.location.hash = alternativeHash
```

Comment:
This is a tiny bit risky because it can lead to infinite redirects.

Reply:
I don't think so, because:

- we only redirect if there is no ID match on the page
- when we redirect we know that a match exists on the page to which we direct to
- so it stops after one iteration

Or do I miss a scenario?